core/src/main/java/org/apache/iceberg/ScanSummary.java

```diff
@@ -159,7 +159,7 @@ private void removeTimeFilters(List<Expression> expressions, Expression expressi
       removeTimeFilters(filters, Expressions.rewriteNot(scan.filter()));
       Expression rowFilter = joinFilters(filters);
 
-      Iterable<ManifestFile> manifests = table.currentSnapshot().manifests();
+      Iterable<ManifestFile> manifests = table.currentSnapshot().dataManifests();
```

Comment:
This whole class will need to be updated to support detecting changes in delete files, so I just did the minimum to update it safely. Since it uses `ManifestFiles.read` later instead of `ManifestFiles.readDeletes`, it made sense to only use data manifests. This is one that we should plan to update in a separate PR, if anyone is using it.